modules/runner/src/integrationTest/java/org/apache/ignite/internal/table/ItInternalTableTest.java

```diff
@@ -384,78 +359,190 @@ public void testRoScanAllExplicitPopulatingData() throws InterruptedException {
         roScanAll(false);
     }
 
-    private void roScanAll(boolean implicit) throws InterruptedException {
-        IgniteImpl node = node();
+    @Test
+    public void getAllOrderTest() throws Exception {
+        var keyRows = populateEvenKeysAndPrepareEntriesToLookup(true);
 
         InternalTable internalTable = ((TableImpl) table).internalTable();
+        var schemaDescriptor = ((TableImpl) table).schemaView().schema();
 
-        KeyValueView<Tuple, Tuple> keyValueView = table.keyValueView();
+        List<BinaryRow> res = internalTable.getAll(keyRows, null).get();
 
-        Publisher<BinaryRow> res = internalTable.scan(0, node.clock().now(), node.node());
+        assertEquals(keyRows.size(), res.size());
 
-        var subscriberAllDataAwaitLatch = new CountDownLatch(1);
+        var resIter = res.iterator();
 
-        var retrievedItems = new ArrayList<BinaryRow>();
+        for (BinaryRowEx key : keyRows) {
+            int i = TableRow.keyTuple(Row.wrapKeyOnlyBinaryRow(schemaDescriptor, key)).<Long>value("key").intValue();
 
-        res.subscribe(new Subscriber<>() {
-            @Override
-            public void onSubscribe(Subscription subscription) {
-                subscription.request(10000);
-            }
+            var resRow = resIter.next();
```

Comment:
It is incorrect to use `var` because the type is not visible from the line of code.